ts_src/confidential.ts

```diff
@@ -65,882 +9,406 @@ export interface UnblindOutputResult {
-    init.inputIndex,
-    inputAssetBlindingFactors[init.inputIndex],
-    outputAssetBlindingFactor,
-  );
-
-  return surjectionproof.serialize(proof);
+interface Generator {
+  generate: (seed: Buffer) => Buffer;
+  generateBlinded(key: Buffer, blind: Buffer): Buffer;
+  parse(input: Buffer): Buffer;
+  serialize(generator: Buffer): Buffer;
 }
 
-export async function surjectionProofVerify(
-  inAssets: Buffer[],
-  inAssetBlinders: Buffer[],
-  outAsset: Buffer,
-  outAssetBlinder: Buffer,
-  proof: Buffer,
-): Promise<boolean> {
-  const { generator, surjectionproof } = await secp256k1Promise;
-  const inGenerators = inAssets.map((v, i) =>
-    generator.generateBlinded(v, inAssetBlinders[i]),
-  );
-  const outGenerator = generator.generateBlinded(outAsset, outAssetBlinder);
-  const sProof = surjectionproof.parse(proof);
-  return surjectionproof.verify(sProof, inGenerators, outGenerator);
+interface Pedersen {
+  commit(blindFactor: Buffer, value: string, generator: Buffer): Buffer;
+  commitSerialize(commitment: Buffer): Buffer;
+  commitParse(input: Buffer): Buffer;
+  blindSum(blinds: Array<Buffer>, nneg?: number): Buffer;
+  verifySum(commits: Array<Buffer>, negativeCommits: Array<Buffer>): boolean;
+  blindGeneratorBlindSum(
+    values: Array<string>,
+    nInputs: number,
+    blindGenerators: Array<Buffer>,
+    blindFactors: Array<Buffer>,
+  ): Buffer;
 }
 
-export async function blindValueProof(
-  value: string,
-  valueCommit: Buffer,
-  assetCommit: Buffer,
-  valueBlinder: Buffer,
-  opts?: RngOpts,
-): Promise<Buffer> {
-  const { generator, pedersen, rangeproof } = await secp256k1Promise;
-
-  const gen = generator.parse(assetCommit);
-  const commit = pedersen.commitParse(valueCommit);
-  const nonce = randomBytes(opts);
-
-  return rangeproof.sign(commit, valueBlinder, nonce, value, gen, value, -1);
-}
-
-export async function blindAssetProof(
-  asset: Buffer,
-  assetCommit: Buffer,
-  assetBlinder: Buffer,
-): Promise<Buffer> {
-  const { generator, surjectionproof } = await secp256k1Promise;
-
-  const nInputsToUse = 1;
-  const maxIterations = 100;
-
-  const init = surjectionproof.initialize(
-    [asset],
-    nInputsToUse,
-    asset,
-    maxIterations,
-    ZERO,
-  );
-
-  const gen = generator.generate(asset);
-  const assetGen = generator.parse(assetCommit);
-
-  const proof = surjectionproof.generate(
-    init.proof,
-    [gen],
-    assetGen,
-    init.inputIndex,
-    ZERO,
-    assetBlinder,
-  );
-
-  return surjectionproof.serialize(proof);
+interface RangeProof {
+  info(proof: Buffer): {
+    exp: number;
+    mantissa: string;
+    minValue: string;
+    maxValue: string;
+  };
+  verify(
+    commit: Buffer,
+    proof: Buffer,
+    generator: Buffer,
+    extraCommit?: Buffer,
+  ): boolean;
+  sign(
+    commit: Buffer,
+    blind: Buffer,
+    nonce: Buffer,
+    value: string,
+    generator: Buffer,
+    minValue?: string,
+    base10Exp?: number,
+    minBits?: number,
```

Comment:
these values are always `1, 0, 52`. Do you think we can take the opportunity to remove them from the interface and use the hardcoded values in the concrete impl(s)?

Reply:
this implies changing `secp` library. Would not do it